lib/modules/manager/gomod/update.ts

```diff
@@ -17,16 +17,15 @@ export function updateDependency({
 }: UpdateDependencyConfig): string | null {
   try {
     logger.debug(`gomod.updateDependency: ${upgrade.newValue}`);
-    const { depName, depType, updateType } = upgrade;
-    if (updateType === 'replacement') {
-      logger.warn('gomod manager does not support replacement updates yet');
-      return null;
-    }
+    const { depType, newName } = upgrade;
+    const fromPackageName = upgrade.depName;
+    // newName will be available for replacement
+    const toPackageName = newName ?? fromPackageName;
     // istanbul ignore if: should never happen
-    if (!depName || !upgrade.managerData) {
+    if (!fromPackageName || !toPackageName || !upgrade.managerData) {
```

Comment:
```suggestion
    if (!fromPackageName || !upgrade.managerData) {
```

toPackageName will be same as fromPackageName if it's falsey